### ISSUE TITLE
problems with gamma rate

### ISSUE THREAD

**Issue by idavydov** [sjspielman/pyvolve#3]
It seems that there's problem with gamma rate distribution in the code.
np.random.gamma which is used in _draw_gamma_rates uses kappa/theta (shape/scale) parameterization, while alpha/beta (shape/rate) is assumed.
Scale is 1/rate.
So I think the proper code should look like this:
```
np.random.gamma(self.alpha, scale = 1./self.alpha, size = self.k_gamma)
```
Another problem is that as far as I can see category rates are drawn randomly from the gamma distribution. As far as I understand pdf function should be split instead into equiprobable chunks, and mean value for each one should computed. This is easy to do using cdf function.

**Comment by sjspielman** [comment]
I'll look into this today and I'll keep you posted.

On Mon, Jan 18, 2016 at 5:21 AM, Iakov Davydov <notifications@github.com>
wrote:

> It seems that there's problem with gamma rate distribution in the code
> nprandomgamma which is used in _draw_gamma_rates uses kappa/theta
> (shape/scale) parameterization, while alpha/beta parameterization is assumed
> The problem is that shape=scale in kappa/theta parameterization doesn't
> give you mean value of 1, while it does in alpha/beta parameterization
> In the first case expected mean value is shape*scale, while in the second
> it's shape/rate
> I think the proper code should look like this:
>
> nprandomgamma(selfalpha, scale = 1/selfalpha, size = selfk_gamma)
>
> Another problem is that as far as I can see category rates are drawn
> randomly from the gamma distribution As far as I understand pdf function
> should be split instead into equiprobable chunks, and mean value for each
> one should computed This is easy to do using cdf function
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sjspielman/pyvolve/issues/3>.
>



-- 
Stephanie J. Spielman
Ph.D. Candidate, Wilke Lab
Department of Integrative Biology
Center for Computational Biology and Bioinformatics
University of Texas at Austin

**Comment by sjspielman** [comment]
The implementation has been addressed - upgrade to 0.7.4! Note that this has been updated in PyPI too.

**Comment by idavydov** [comment]
Thanks a lot for fixing this.
